### PR TITLE
US461402 - Code changes to support special characters in Parsing Tag …

### PR DESCRIPTION
**Pull request by SharedEagles-Pairing**
Description - Code changes to support special characters in Parsing Tag Expressions
User Story: US461402
Note: Please tag with version 1.1.6

### CODE REVIEW

File: tagExpression/utils.js [hunk 1]
```diff
@@ -1,23 +1,136 @@
...
+    let formulaForTheLexerParser = '';
+    let lastIndexReplaced = 0;
+
+    // eslint-disable-next-line no-useless-escape
+    const specialCharRegex = /[\#\$\.\,\+\?\|\&\-\_\=\>\^\@\(\)\[\]\\\/\'\%\!\*\`\{\}\~\<\"\:\;]/g;
+
+    const lengthOfMatchedGroups = listOfMatchedGroups.length;
+
+    let allTagsValid = true;
+    listOfMatchedGroups.every((currentValue, currentIndex) => {
```
Comment (SharedEagles-Pairing) [inline tagExpression/utils.js]: On the backend, do we need to check if any asset-tags matches present as well ?
i.e lengthOfMatchedGroups > 0

Comment (SharedEagles-Pairing) [reply tagExpression/utils.js]: Added the check

**Review by vibhaMalviya — APPROVED** [review]

File: tagExpression/utils.js [hunk 2]
```diff
@@ -1,23 +1,143 @@
...
+ * 		lexSafeFormula: string,
+ * 		tagNameToUniqueIDMap:Map<string, string>,
+ * 		allTagsValid: boolean	
+ *  }
+ * tagNameToUniqueIDMap : key -> Tag Identifier is the special character trimmed in the original Tag
+ * tagNameToUniqueIDMap : value -> Mapped TagName
+ *
+ * @private
+ */
+function getLexParserSafeTagFormula(formula, tagNames) {
```
Comment (vibhaMalviya) [inline tagExpression/utils.js:23]: Same method has been added to common-service. Are you planning to use this method from there?

File: tagExpression/utils.js [hunk 3]
```diff
@@ -1,23 +1,143 @@
...
- * Get parsed 'srcKeyFormula' - using 'LexerParser'
+ * Get parsed 'srcKeyFormula' - using 'LexerParser' includes support for special characters in tag names
  * created this fn to address 'DE146572'
- * @param formula
- * @private
+ * @param formula, tagNames
+ * @returns srcKeyFormula (parsed Formula)
  */
-function getParsedSrcKeyFormula(formula) {
+function getParsedSrcKeyFormula(formula, tagNames) {
```
Comment (vibhaMalviya) [inline tagExpression/utils.js:104]: same as above comment

**Pull request merged**